Lib/glyphsLib/builder/features.py

```diff
@@ -138,7 +138,11 @@ def _to_ufo_features(
                     # Replace special chars backslash and doublequote for AFDKO syntax
                     name = name.replace("\\", r"\005c").replace('"', r"\0022")
                     feature_names = ["featureNames {", f'  name "{name}";', "};"]
-        elif font.format_version == 3 and feature.labels:
+        elif (
+            font.format_version == 3
+            and feature.labels
+            and all(l["value"] != "" for l in feature.labels)
```

Comment:
maybe you meant `any(l["value"] != "" for l in feature.labels)`?
I think we want to skip the whole `featureNames` section if all the labels are empty (or add one if any is non empty)

Also below in the `for label in feature.labels:` loop, I think you want to do `continue` if the name == ""

Reply:
I've decided to drop `any` or `all`, far easier to just include them if they get appended to the `feature_names` list.